Determine if a URL is a local scheme

rreno opened the pull request:
#### 64e3632fceb74495f42efe3118ca243254c422c8
<pre>
Determine if a URL is a local scheme
<a href="https://bugs.webkit.org/show_bug.cgi?id=244403">https://bugs.webkit.org/show_bug.cgi?id=244403</a>
rdar://problem/99286879

Reviewed by Darin Adler.

The HTML spec defines local schemes as about, blob, or data. This adds a
member function to URL to query whether or not a URL instance is local.

This is a shorthand for url.protocolIsAbout() || url.protocolIsBlob() || url.protocolIsData().

* Source/WTF/wtf/URL.cpp:
(WTF::URL::hasLocalScheme const):
* Source/WTF/wtf/URL.h:

Canonical link: <a href="https://commits.webkit.org/253928@main">https://commits.webkit.org/253928@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bf577123dce61e5a5210058898778c5147200427

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/87476 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/31565 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/18277 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/96702 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/150157 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/29929 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/26108 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/79591 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/91471 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/93095 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/68/builds/24181 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/74258 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/24017 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/79148 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/79366 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/67035 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/79315 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/27638 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/13198 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/72996 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/27590 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/14214 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/25991 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2760 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/29280 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/37079 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/75824 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/29206 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/33492 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/16809 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler reviewed (approved):
This change is OK, but normally we would combine it with at least one use of the new function, and a test covering that use, so the function would be tested.

Alternatively we could include a unit test.

When the function is as simple as this, it’s helpful to do a single pull request that combines the addition of the function with the first use.

rreno committed:
Determine if a URL is a local scheme
https://bugs.webkit.org/show_bug.cgi?id=244403
rdar://problem/99286879

Reviewed by Darin Adler.

The HTML spec defines local schemes as about, blob, or data. This adds a
member function to URL to query whether or not a URL instance is local.

This is a shorthand for url.protocolIsAbout() || url.protocolIsBlob() || url.protocolIsData().

* Source/WTF/wtf/URL.cpp:
(WTF::URL::hasLocalScheme const):
* Source/WTF/wtf/URL.h:

Canonical link: https://commits.webkit.org/253928@main
